rclcpp_generic/test/rclcpp_generic/test_typesupport_helpers.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "rcpputils/shared_library.hpp"
+
+#include "rclcpp_generic/typesupport_helpers.hpp"
+
+using namespace ::testing;  // NOLINT
+
+TEST(TypesupportHelpersTest, throws_exception_if_filetype_has_no_type) {
+  EXPECT_ANY_THROW(rclcpp_generic::extract_type_identifier("just_a_package_name"));
+}
+
+TEST(TypesupportHelpersTest, throws_exception_if_filetype_has_slash_at_the_start_only) {
+  EXPECT_ANY_THROW(rclcpp_generic::extract_type_identifier("/name_with_slash_at_start"));
+}
+
+TEST(TypesupportHelpersTest, throws_exception_if_filetype_has_slash_at_the_end_only) {
```

Comment:
Is there a document somewhere that explains the rationale for having both `pkg/type` and `pkg/msg/type`?

Reply:
That would be useful for correctly implementing further tests

Reply:
I'm not 100% sure but i think it's for "brain-backwards-compatibility" in to match ROS 1 naming convention?

Reply:
Do you know why the new convention was introduced in the first place?